meshio/_files.py

```diff
@@ -0,0 +1,22 @@
+import os
+import sys
+from contextlib import contextmanager
+
+
+def is_buffer(obj, mode):
+    return ("r" in mode and hasattr(obj, "read")) or (
+        "w" in mode and hasattr(obj, "write")
+    )
+
+
+@contextmanager
+def open_file(path_or_buf, mode="r"):
+    if is_buffer(path_or_buf, mode):
+        yield path_or_buf
+    elif sys.version_info < (3, 6) and isinstance(path_or_buf, os.PathLike):
+        # todo: remove when 3.5 is EoL
```

Comment:
Best explicitly say 2020-09-13 here and link to https://devguide.python.org/#status-of-python-branches